tfjs-backend-webgpu/src/matmul_test.ts

```diff
@@ -17,9 +17,27 @@
 
 import * as tf from '@tensorflow/tfjs-core';
 import {test_util} from '@tensorflow/tfjs-core';
-import {describeWebGPU} from './test_util';
+// tslint:disable-next-line: no-imports-from-dist
+import {describeWithFlags} from '@tensorflow/tfjs-core/dist/jasmine_util';
+
+import {MATMUL_ENVS} from './backend_webgpu_test_registry';
+import {MatMulProgramType} from './webgpu_util';
+
+const {expectArraysClose} = test_util;
+const MATMUL_SHARED_DIM_THRESHOLD = 1000;
+let type0 = MatMulProgramType.MatMulReduceProgram;
+let type1 = MatMulProgramType.MatMulReduceProgram;
+const matmulTest = () => {
+  let savedMatmulFlag = -1;
+  beforeAll(() => {
+    savedMatmulFlag = tf.env().get('WEBGPU_TESTED_MATMUL') as number;
+    tf.env().set('WEBGPU_TESTED_MATMUL', type0);
+    type0++;
+  });
+  afterAll(() => {
+    tf.env().set('WEBGPU_TESTED_MATMUL', savedMatmulFlag);
+  });
```

Comment:
This was missing from my last review:

Instead of setting WEBGPU_TESTED_MATMUL to type0 and incrementing it each time `matmulTest` is called, it would be better to pass the MatMulProgramType in a closure.

```ts
function matmulTest(programType: MatMulProgramType) {
  return () => {
    let savedMatmulFlag = -1;
    beforeAll(() => {
      savedMatmulFlag = tf.env().get('WEBGPU_TESTED_MATMUL') as number;
      tf.env().set('WEBGPU_TESTED_MATMUL', programType);
    });
    afterAll(() => {
      tf.env().set('WEBGPU_TESTED_MATMUL', savedMatmulFlag);
    });
    ...
  }
}
```

Reply:
Thanks @mattsoulanille . Great suggestions! Previously, I use `function matmulTest(programType: MatMulProgramType) {...}` but didn't have `return () => {}` to wrap the body. It will report an error to me. I thought I couldn't pass the type. So I added an ugly global variable value. Your suggestion looks better.
I have addressed all your comments. Pease take another look, thanks.